src/urbit/app/chess.hoon

```diff
@@ -749,25 +773,61 @@
       %+  turn  ids
       |=  a=@dau
       [(scot %da a) ~]
+    ::
+    ::  .^(noun %gx /=chess=/friends/noun)
+    ::  .^(json %gx /=chess=/friends/json)
+    ::  read mutual friends
+    [%x %friends ~]
+      ``[%chess-pals !>((~(mutuals pals bowl) ~.))]
   ==
 ++  on-agent
   |=  [=wire =sign:agent:gall]
   ^-  (quip card _this)
   ?+  wire  (on-agent:default wire sign)
     ::
     ::  remove a sent challenge on nack
-    [%poke %challenge ~]
+    [%poke %challenge %send ~]
       ?+  -.sign  (on-agent:default wire sign)
         %poke-ack
+          ::  if opponent has acked our challenge, add it to the frontned
           ?~  p.sign
-            [~ this]
+          ::  
+            :-  :~  :*  %give  %fact   ~[/challenges]
+                        %chess-update  !>([%challenge-sent src.bowl u.+:(~(get by challenges-sent) src.bowl)])
+                    ==
+                ==
+            this
+          ::
+          ::  XX: send a notification to the user that an error occurred
+          ::
+          ::  if not, print the error and
+          ::  consider the challenge declined
           %-  (slog u.p.sign)
-          :-  ~
+          :-  :~  :*  %give  %fact   ~[/challenges]
+                      %chess-update  !>([%challenge-declined src.bowl])
+                  ==
+              ==
           %=  this
             challenges-sent  (~(del by challenges-sent) src.bowl)
           ==
       ==
     ::
+    :: get ack/nack when we accept/reject a challenge
+    [%poke %challenge %resolve ~]
+      ?+  -.sign  (on-agent:default wire sign)
+        %poke-ack
+          ?~  p.sign
+          ::  if opponent has acked our acceptance/rejection,
```

Comment:
This case is only for when our opponent acks a rejection, not acceptance. We're forgetting to send `%challenge-resolved` cards when the opponent accepts our game (i.e. subscribes to us): after lines `664` and `698` in the code for `[%game @ta %moves ~]` in `on-watch`.